partner/WHMCS/modules/servers/jetpack/jetpack.php

```diff
@@ -204,31 +212,52 @@ function make_api_request($url, $auth='', $data=[])
 function save_next_url($url, $params)
 {
     $jetpack_next_url_field = Capsule::table('tblcustomfields')
-        ->where(array('fieldname' => 'jetpack_next_url', 'type' => 'product'))->first();
+        ->where(array('fieldname' => 'jetpack_provisioning_details', 'type' => 'product'))->first();
 
     Capsule::table('tblcustomfieldsvalues')->insert(array(
         'fieldid' => $jetpack_next_url_field->id, 'relid' => $params['model']['orderId'], 'value' => $url));
 }
 
 /**
- * Validate that the fields required to provision a Jetpack plan are present
- * and valid for those that can be verified
+ * Validate that the module was correctly set up when the product was
+ * created by the WHMCS user and that the required Fields/Options for
+ * being able to provision a Jetpack plan are present. Fields validated are
+ *  - Allowed Plans from Plan Custom Field
+ *  - Required Custom Fields
+ *  - Required Config Options
  *
  * @param array $params
  * @return bool
+ * @throws Exception
  */
 function validate_required_fields(array $params)
 {
     $allowed_plans = array('free', 'personal', 'premium', 'professional');
+    $required_custom_fields = array('Plan', 'Site URL', 'Local User');
+
+    foreach ($required_custom_fields as $field)
+        if (!isset($params['customfields'][$field])) {
+            throw new Exception('The module does not appear to be setup correctly. The required custom field '
```

Comment:
What's the trade-off of using exceptions in `validate_required_fields` instead of returning errors?

Reply:
Exceptions were previously used so they could be caught and used in logModuleCall but based on the feedback you previously provided above about better logging functions and further digging in to WHMCS I've updated validation to now return error strings which will automatically be logged in the Activity Logs if they are returned.